MAINTAINERS.yml

```diff
@@ -4743,6 +4743,17 @@ West:
   labels:
     - "platform: Intel"
 
+"West project: libmctp":
+  status: maintained
+  maintainers:
+    - teburd
+  collaborators:
+    - nashif
```

Comment:
Hi, Tom

Can you add @inteljiangwe1 who as collaborator?
Wei is our owner on MCTP, who can help work on this module, and add other components.

BR,
Ke Han